Source/WebKit/GPUProcess/graphics/WebGPU/RemoteBuffer.cpp

```diff
@@ -98,21 +98,7 @@ void RemoteBuffer::copy(std::optional<WebCore::SharedMemoryHandle>&& dataHandle,
         completionHandler(false);
         return;
     }
-
-    auto buffer = m_backing->getBufferContents();
-    if (buffer.empty()) {
-        completionHandler(false);
-        return;
-    }
-
-    auto endOffset = checkedSum<size_t>(offset, data.size());
-    if (endOffset.hasOverflowed() || endOffset.value() > buffer.size()) {
-        completionHandler(false);
-        return;
-    }
-
-    memcpySpan(buffer.subspan(offset), data);
-    completionHandler(true);
```

Comment:
Here again, we want to avoid changing the structure / design as much as possible when we refactor. 

Let's keep the existing structure, and just sub in the other copy function if we're building with Swift.

Reply:
With this PR we are only enabling Swift interop at the WebGPU.framework level. Not at the WebKit Level. So I can't stub out the Swift API here.
The intent here is to move the copy operation down to that level. So, while doing that I am also moving bounds checking down so that we check at the same level we call memcopy, it's just more coherent.